contracts/LiquidDemocracy.sol

```diff
@@ -54,4 +57,22 @@ contract LiquidDemocracy {
         votersData[voterAddress].weight = 1;
         votersData[voterAddress].registered = true;
     }
+
+    /**
+    * @notice Record a vote in the ballot
+    * @param voter The address of the voter
+    * @param voteOption The integer option chosen by voter
+    */
+    function vote(address voter, uint voteOption) public {
+        // Cannot vote more than once and must be registered
+        require(votersData[voter].registered == true);
+        require(votersData[voter].voted != true);
```

Comment:
replace with:
 require(votersData[voter].registered);
and
require(!votersData[voter].voted);